build.md

```diff
@@ -227,6 +227,28 @@ Cache target is defined using the same `type=TYPE[,KEY=VALUE]` syntax defined by
 
 Unsupported cache target MUST be ignored and not prevent user from building image.
 
+### additional_contexts
+
+`additional_contexts` defines a list of named contexts the image builder SHOULD use during image build.
+
+`additional_contexts` can be a mapping or a list:
+
+```yml
+build:
+  context: .
+  additional_contexts:
+    - project=/path/to/context
```

Comment:
Might be a good idea to include a couple examples of other "types", i.e. Git URLs, `docker-image://`

We could even specifically call out that the values are strings but validation beyond that is builder-specific (e.g. even within BuildKit, it'll depend on the version whether it supports it)